l3_dump.py

```diff
@@ -500,7 +500,11 @@ def do_main(args:list, return_logentry_lists:bool = False):
             # print(f"{msgptr=:x}, {fibase=:x}, {rodata_offs=:x}, {offs=}")
 
             # Generate C-style sprintf() output on message-string.
-            msg_text = do_c_print(strings[offs], arg1, arg2)
+            try:
+                msg_text = do_c_print(strings[offs], arg1, arg2)
+            except KeyError:
```

Comment:
This change _seems_ reasonable, except in the callee, I see this comment block:

```
382     If there is any exception in printing, fall-back to just dumping the
383     input arguments, without any modifications.
```

This fall-back behavior I had coded / intended for now conflicts with the new error raising you are implementing.

My thinking in coding the fall-back behavior was that in the user's program, we should expect a properly formatted print format-string. So, what's the likelihood of getting an error?

I don't know how or where KeyError exception will be raised.

I am not sure how to sort out what you are trying to do v/s what I had coded earlier.

---
I will guide you to the pytests in `pytests/l3_dump_parse_test.py`. 

See test_fmtstr_replace() and test_fmtstr_replace_and_print().

From code coverage point of view, can you investigate if it's possible to add a new pytest invoking `do_c_print()`, i.e. add `test_do_c_print()`, ... and see if there is some sense in testing out by invoking an exception raising situation.

I don't have time to sort out the pytest angle. It's just an idea for you to follow-thru as best possible.

Reply:
Hmm, I see what you're saying, I think you're right, this should never be user visible. I'll drop this commit from the PR.